### PR TITLE
Update vm-common and vms + remove old methods part 1

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- integrate new interface
  
## Proposed changes
- methods replaced:
1. IsSCDeployFlagEnabled()
2. IsBuiltInFunctionsFlagEnabled()
3. IsRelayedTransactionsFlagEnabled()
4. IsPenalizedTooMuchGasFlagEnabled()
5. ResetPenalizedTooMuchGasFlag()
6. IsSwitchJailWaitingFlagEnabled()
7. IsBelowSignedThresholdFlagEnabled()
8. IsSwitchHysteresisForMinNodesFlagEnabled()
9. IsSwitchHysteresisForMinNodesFlagEnabledForCurrentEpoch()
10. IsTransactionSignedWithTxHashFlagEnabled()
11. IsMetaProtectionFlagEnabled()
12. IsAheadOfTimeGasUsageFlagEnabled()
13. IsGasPriceModifierFlagEnabled()
14. IsRepairCallbackFlagEnabled()
15. IsBalanceWaitingListsFlagEnabled()
16. IsReturnDataToLastTransferFlagEnabled()
17. IsSenderInOutTransferFlagEnabled()
18. IsStakeFlagEnabled()
19. IsStakingV2FlagEnabled()
20. IsStakingV2OwnerFlagEnabled()
21. IsStakingV2FlagEnabledForActivationEpochCompleted()

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: epochStart/metachain/systemSCs.go [hunk 1]
```diff
@@ -178,14 +178,15 @@ func (s *systemSCProcessor) ProcessSystemSmartContract(
 	nonce uint64,
 	epoch uint32,
 ) error {
-	if s.enableEpochsHandler.IsSwitchHysteresisForMinNodesFlagEnabledForCurrentEpoch() {
+	currentEpoch := s.enableEpochsHandler.GetCurrentEpoch()
+	if s.enableEpochsHandler.IsSwitchHysteresisForMinNodesFlagEnabledInSpecificEpochOnly(currentEpoch) {
```
Comment (ssd04) [inline epochStart/metachain/systemSCs.go:182]: from what i see we use current epoch in all calls, should we think of getting current epoch inside enableEpochsHandler component as default behavior or something?

Comment (ssd04) [reply epochStart/metachain/systemSCs.go:182]: or there are other flags that will use a different epoch? and will be updated in a next PR

Comment (iulianpascalau) [reply epochStart/metachain/systemSCs.go:182]: that was the main change for the enable epochs handler. The previous implementation did this internally. :D 
The refactoring is needed so we can ask the component if a particular flag is active or not depending on a provided epoch. Real case usage: we have a component that will need to answer the status on some flags for older epochs.

File: process/transaction/shardProcess_test.go [hunk 2]
```diff
@@ -88,11 +90,11 @@ func createArgsForTxProcessor() txproc.ArgsNewTxProcessor {
 		ArgsParser:       &mock.ArgumentParserMock{},
 		ScrForwarder:     &mock.IntermediateTransactionHandlerMock{},
 		EnableEpochsHandler: &enableEpochsHandlerMock.EnableEpochsHandlerStub{
-			IsPenalizedTooMuchGasFlagEnabledField: true,
+			IsPenalizedTooMuchGasFlagEnabledInEpochCalled: func(epoch uint32) bool { return true },
```
Comment (iulianpascalau) [inline process/transaction/shardProcess_test.go:93]: ```IsPenalizedTooMuchGasFlagEnabledInEpochCalled: flagActiveTrueHandler``` ?

File: process/transaction/shardProcess_test.go [hunk 3]
```diff
@@ -2090,7 +2092,7 @@ func TestTxProcessor_ProcessRelayedTransaction(t *testing.T) {
 	args.PubkeyConv = pubKeyConverter
 	args.ArgsParser = smartContract.NewArgumentParser()
 	args.EnableEpochsHandler = &enableEpochsHandlerMock.EnableEpochsHandlerStub{
-		IsRelayedTransactionsFlagEnabledField: true,
+		IsRelayedTransactionsFlagEnabledInEpochCalled: func(epoch uint32) bool { return true },
```
Comment (iulianpascalau) [inline process/transaction/shardProcess_test.go:2095]: `IsRelayedTransactionsFlagEnabledInEpochCalled: flagActiveTrueHandler`

File: process/smartContract/processProxy/processProxy_test.go [hunk 4]
```diff
@@ -2,12 +2,13 @@ package processProxy
...
 import (
 	"fmt"
-	"github.com/multiversx/mx-chain-go/testscommon/enableEpochsHandlerMock"
 	"math/big"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/multiversx/mx-chain-go/testscommon/enableEpochsHandlerMock"
+
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply process/smartContract/processProxy/processProxy_test.go:11]: removed line, import is still needed for the stub

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**